Tribler/Policies/BoostingManager.py

```diff
@@ -0,0 +1,493 @@
+# -*- coding: utf-8 -*-
+# Written by Egbert Bouman, Mihai Capotă, Elric Milon, and Ardhi Putra Pratama H
+"""Manage boosting of swarms"""
+import logging
+import os
+import shutil
+from binascii import hexlify, unhexlify
+
+import libtorrent as lt
+from twisted.internet.task import LoopingCall
+
+from Tribler.Core.DownloadConfig import DownloadStartupConfig, DefaultDownloadStartupConfig
+from Tribler.Core.Libtorrent.LibtorrentDownloadImpl import LibtorrentDownloadImpl
+from Tribler.Core.Utilities import utilities
+from Tribler.Core.Utilities.configparser import CallbackConfigParser
+from Tribler.Core.exceptions import OperationNotPossibleAtRuntimeException
+from Tribler.Core.simpledefs import DLSTATUS_SEEDING, NTFY_TORRENTS, NTFY_UPDATE, NTFY_CHANNELCAST
+from Tribler.Policies.BoostingPolicy import SeederRatioPolicy
+from Tribler.Policies.BoostingSource import ChannelSource
+from Tribler.Policies.BoostingSource import DirectorySource
+from Tribler.Policies.BoostingSource import RSSFeedSource
+from Tribler.Policies.credit_mining_util import source_to_string, string_to_source, compare_torrents
+from Tribler.Policies.defs import SAVED_ATTR, CREDIT_MINING_FOLDER_DOWNLOAD, CONFIG_KEY_ARCHIVELIST, \
+    CONFIG_KEY_SOURCELIST, CONFIG_KEY_ENABLEDLIST, CONFIG_KEY_DISABLEDLIST
+from Tribler.dispersy.taskmanager import TaskManager
+
+
+class BoostingSettings(object):
+    """
+    Class contains settings on boosting manager
+    """
+    def __init__(self, session, policy=SeederRatioPolicy, load_config=True):
+        self.session = session
+
+        # Configurable parameter (changeable in runtime -plus sources-)
+        self.max_torrents_active = 20
+        self.max_torrents_per_source = 10
+        self.source_interval = 100
+        self.swarm_interval = 100
+
+        # Can't be changed in runtime
+        self.tracker_interval = 200
+        self.logging_interval = 60
+        self.share_mode_target = 3
+        self.policy = policy(session)
+
+        # Non-Configurable
+        self.initial_logging_interval = 20
+        self.initial_tracker_interval = 25
+        self.initial_swarm_interval = 30
+        self.min_connection_start = 5
+        self.min_channels_start = 100
+        self.credit_mining_path = os.path.join(DefaultDownloadStartupConfig.getInstance().get_dest_dir(),
+                                               CREDIT_MINING_FOLDER_DOWNLOAD)
+        self.load_config = load_config
+
+
+class BoostingManager(TaskManager):
+    """
+    Class to manage all the credit mining activities
+    """
+
+    __single = None
+
+    def __init__(self, session, settings=None):
+        super(BoostingManager, self).__init__()
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+        BoostingManager.__single = self
+        self.boosting_sources = {}
+        self.torrents = {}
+
+        self.session = session
+        assert self.session.get_libtorrent()
```

Comment:
should be implemented in https://github.com/Tribler/tribler/pull/2064/commits/e02ac24d16ec097c3297ff4cd4624cfdecb99f96